File: app/src/main/java/com/coinninja/coinkeeper/view/widget/PaymentBarView.java
package com.coinninja.coinkeeper.view.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;

import com.coinninja.coinkeeper.R;

public class PaymentBarView extends ConstraintLayout {

    private OnScanPressedObserver scanPressedObserver;
    private OnRequestPressedObserver requestPressedObserver;
    private OnSendPressedObserver sendPressedObserver;

    public PaymentBarView(Context context) {
        this(context, null);
    }

    public PaymentBarView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public PaymentBarView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    public void setOnRequestPressedObserver(OnRequestPressedObserver requestPressedObserver) {
        this.requestPressedObserver = requestPressedObserver;
    }

    public void setOnSendPressedObserver(OnSendPressedObserver sendPressedObserver) {
        this.sendPressedObserver = sendPressedObserver;
    }

    public void setOnScanPressedObserver(OnScanPressedObserver scanPressedObserver) {
        this.scanPressedObserver = scanPressedObserver;
    }

    private void init(Context context) {
        LayoutInflater.from(context).inflate(R.layout.merge_component_payment_bar, this, true);
        findViewById(R.id.scan_btn).setOnClickListener(this::onScanPressed);
        findViewById(R.id.send_btn).setOnClickListener(this::onSendPressed);
        findViewById(R.id.request_btn).setOnClickListener(this::onRequestPressed);
    }

    private void onRequestPressed(View view) {
        if (requestPressedObserver != null) requestPressedObserver.onRequestPressed();
    }

    private void onSendPressed(View view) {
        if (sendPressedObserver != null) sendPressedObserver.onSendPressed();
    }

    private void onScanPressed(View view) {
        if (scanPressedObserver != null) scanPressedObserver.onScanPressed();
    }

    public interface OnSendPressedObserver {
        void onSendPressed();
    }

    public interface OnScanPressedObserver {
        void onScanPressed();
    }

    public interface OnRequestPressedObserver {
        void onRequestPressed();
    }

}

File: app/src/main/java/com/coinninja/coinkeeper/view/widget/PaymentBarView.kt
package com.coinninja.coinkeeper.view.widget

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import androidx.constraintlayout.widget.ConstraintLayout
import com.coinninja.android.helpers.styleAsBitcoin
import com.coinninja.android.helpers.styleAsLightning
import com.coinninja.coinkeeper.R
import com.coinninja.coinkeeper.R.layout
import com.coinninja.coinkeeper.cn.wallet.mode.AccountMode

class PaymentBarView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : ConstraintLayout(context, attrs, defStyleAttr) {
    var scanPressedObserver: OnScanPressedObserver? = null
    var requestPressedObserver: OnRequestPressedObserver? = null
    var sendPressedObserver: OnSendPressedObserver? = null

    val scanButton: Button get() = findViewById(R.id.scan_btn)
    val sendButton: Button get() = findViewById(R.id.send_btn)
    val requestButton: Button get() = findViewById(R.id.request_btn)
    var accountMode: AccountMode = AccountMode.BLOCKCHAIN
        set(value) {
            field = value
            invalidateButtonSkins()
        }

    private fun invalidateButtonSkins() {
        when (accountMode) {
            AccountMode.LIGHTNING -> {
                findViewById<Button>(R.id.send_btn).styleAsLightning()
                findViewById<Button>(R.id.request_btn).styleAsLightning()
            }
            else -> {
                findViewById<Button>(R.id.send_btn).styleAsBitcoin()
                findViewById<Button>(R.id.request_btn).styleAsBitcoin()
            }
        }

    }

    fun setOnRequestPressedObserver(requestPressedObserver: OnRequestPressedObserver?) {
        this.requestPressedObserver = requestPressedObserver
    }

    fun setOnSendPressedObserver(sendPressedObserver: OnSendPressedObserver?) {
        this.sendPressedObserver = sendPressedObserver
    }

    fun setOnScanPressedObserver(scanPressedObserver: OnScanPressedObserver?) {
        this.scanPressedObserver = scanPressedObserver
    }

    private fun init(context: Context) {
        LayoutInflater.from(context).inflate(layout.merge_component_payment_bar, this, true)
        scanButton.setOnClickListener { scanPressedObserver?.onScanPressed() }
        sendButton.setOnClickListener { sendPressedObserver?.onSendPressed() }
        requestButton.setOnClickListener { requestPressedObserver?.onRequestPressed() }
    }


    interface OnSendPressedObserver {
        fun onSendPressed()
    }

    interface OnScanPressedObserver {
        fun onScanPressed()
    }

    interface OnRequestPressedObserver {
        fun onRequestPressed()
    }

    init {
        init(context)
    }
}